rules/opentitan.bzl

```diff
@@ -825,6 +827,14 @@ def opentitan_binary(
         testonly = testonly,
         **kwargs
     )
+    elf_transition_binary_name = "{}_elf_transition".format(name)
+    targets.append(":" + elf_transition_binary_name)
+    platform_target(
+        name = elf_transition_binary_name,
+        platform = platform,
+        target = native_binary_name,
+        testonly = testonly,
+    )
```

Comment:
Would it work if we make the `<name>.elf` target with transition and make the cc_binary target without transition internal?

Reply:
In principle yes but with the complex set of macros in `opentitan.bzl`, it's anyone's guess if this going to break things or not. I'll have a look.

Reply:
I had a try and it breaks too many things. The reason is not unexpected: several consumers to the elf file expect the elf target to be a cc_binary, ie export a number of providers that cc_binary creates (e.g. CcInfo). At the moment, the `platform_target` rule does not forward any provider but `DefaultInfo`. This means that we either 

- stick with the current solution
- have to export the `cc_binary` output but change its name to something else and change all the users to use this new name (not great)
- modify `platform_target` to forward more providers.

I think the third solution is clearly preferable if possible but I don't know how feasible it actually is. I'll have a quick try to see how it looks.